### PR TITLE
JDK-8300264: Remove metaprogramming/isPointer.hpp

### PR DESCRIPTION
**Pull request by jcking**
Code cleanup of pre-C++11 <type_traits> implementations.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300264](https://bugs.openjdk.org/browse/JDK-8300264): Remove metaprogramming/isPointer.hpp

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12041/head:pull/12041` \
`$ git checkout pull/12041`

Update a local copy of the PR: \
`$ git checkout pull/12041` \
`$ git pull https://git.openjdk.org/jdk pull/12041/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12041`

View PR using the GUI difftool: \
`$ git pr show -t 12041`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12041.diff">https://git.openjdk.org/jdk/pull/12041.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/utilities/concurrentHashTable.inline.hpp [hunk 1]
```diff
@@ -502,7 +504,7 @@ inline void ConcurrentHashTable<CONFIG, F>::
     Bucket* prefetch_bucket = (bucket_it+1) < stop_idx ?
                               table->get_bucket(bucket_it+1) : NULL;
 
-    if (!HaveDeletables<IsPointer<VALUE>::value, EVALUATE_FUNC>::
+    if (!HaveDeletables<std::is_pointer<VALUE>::value, EVALUATE_FUNC>::
         have_deletable(bucket, eval_f, prefetch_bucket)) {
         // Nothing to remove in this bucket.
         continue;
```
Comment (kimbarrett) [inline src/hotspot/share/utilities/concurrentHashTable.inline.hpp:510]: [pre-existing] The if-body is mis-indented.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Review by tschatzl — APPROVED** [review]

**Pull request closed without merging**